node/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -263,8 +275,19 @@ getWalletTxHistory cWalId = toList <<$>> preview (wsTxHistory . ix cWalId)
 getWalletUtxo :: Query Utxo
 getWalletUtxo = view wsUtxo
 
+getWalletBalancesAndUtxo :: Query (WalletBalances, Utxo)
+getWalletBalancesAndUtxo = (,) <$> view wsBalances <*> view wsUtxo
+
+updateWalletBalancesAndUtxo :: UtxoModifier -> Update ()
+updateWalletBalancesAndUtxo modifier = do
+    balAndUtxo <- (,) <$> use wsBalances <*> use wsUtxo
+    wsBalances .= applyUtxoModToAddrCoinMap modifier balAndUtxo
+    wsUtxo %= MM.modifyMap modifier
```

Comment:
Observation: let's remember the fact that `(%=)` uses the lazy variant of `modify` when it comes to `MonadState`, which might or might not be what we want here.